lib/zipkin-tracer/zipkin_tracer_base.rb

```diff
@@ -26,7 +26,7 @@ def with_new_span(trace_id, name)
     end
 
     def may_flush(span)
-      size = spans.values.map(&:size).inject(:+) || 0
+      size = spans.values.map(&:size).map(&:to_i).inject(:+) || 0
```

Comment:
- why would `size` return anything else than a integer
- map => map => inject is a lot of loops inside loops, allocating more objects so IMO this should be a refactored in a good ol' loop where you'd keep track of the total yourself